examples/step-68/step-68.cc

```diff
@@ -0,0 +1,810 @@
+    // describe the details of the particle tracking simulation and its
+    // discretization. The following parameters are about where output should
+    // land, the spatial discretization of the velocity (the default is $Q_1$),
+    // the time step and the output frequency (how many time steps should
+    // elapse before we generate graphical output again):
+    std::string output_directory = "./";
+
+    unsigned int velocity_degree       = 1;
+    double       time_step             = 0.002;
+    double       final_time            = 4.0;
+    unsigned int output_frequency      = 10;
+    unsigned int repartition_frequency = 5;
+
+    // We allow every grid to be refined independently. In this tutorial, no
+    // physics is resolved on the fluid grid, and its velocity is calculated
+    // analytically.
+    unsigned int fluid_refinement              = 4;
+    unsigned int particle_insertion_refinement = 3;
+  };
+
+  // There remains the task of declaring what run-time parameters we can accept
+  // in input files. Since we have a very limited number of parameters, all
+  // parameters are declared in the same section.
+  ParticleTrackingParameters::ParticleTrackingParameters()
+    : ParameterAcceptor("Particle Tracking Problem/")
+  {
+    add_parameter(
+      "Velocity degree", velocity_degree, "", this->prm, Patterns::Integer(1));
+
+    add_parameter("Output frequency",
+                  output_frequency,
+                  "Iteration frequency at which output results are written");
+
+    add_parameter("Repartition frequency",
+                  repartition_frequency,
+                  "Iteration frequency at which the mesh is load balanced");
+
+    add_parameter("Output directory", output_directory);
+
+    add_parameter("Time step", time_step);
+
+    add_parameter("Final time", final_time, "End time of the simulation");
+
+    add_parameter("Fluid refinement",
+                  fluid_refinement,
+                  "Refinement level of the fluid domain");
+
+    add_parameter(
+      "Particle insertion refinement",
+      particle_insertion_refinement,
+      "Refinement of the volumetric mesh used to insert the particles");
+  }
+
+  // @sect3{Velocity profile}
+
+  // The velocity profile is provided as a Function object.
+  // This function is hard-coded within
+  // the example.
+  template <int dim>
+  class Vortex : public Function<dim>
+  {
+  public:
+    Vortex()
+      : Function<dim>(dim)
+    {}
+    virtual void vector_value(const Point<dim> &point,
+                              Vector<double> &  values) const override;
+  };
+
+  template <int dim>
+  void Vortex<dim>::vector_value(const Point<dim> &point,
+                                 Vector<double> &  values) const
+  {
+    const double T = 4;
+    // The velocity profile for the Rayleigh-Kothe vertex is time-dependent.
+    // Consequently, the current time in the
+    // simulation (t) must be gathered from the Function object.
```

Comment:
it works better (=makes it easier to read) if you move the comment above the entire function, because then the whole function shows up in the same highlighted block in the html page